sourcegraph/sourcegraph
# vsce: implement search UI

Implement search UI in the VS Code extension.

Part of post-release cleanup tracked in #32674. 

NOTE: Only need review on changes _outside of_ `client/vscode`. We made [this decision](https://docs.google.com/document/d/1rC51DTAsiza2cnL1BYMEkgNxvISWnsDCmLOtnuI4vhM/edit#bookmark=id.4gncqez405s8) shortly after releasing v1 of the new VS Code extension. We do not have an automated release process yet, and the core codebase is mostly unaffected. The new GitHub Pull Request File Tree should help with this review :).

<details>
<summary>`search` + `search-ui` changes</summary>
<p>Mostly import updates</p>
<img src="https://user-images.githubusercontent.com/37420160/162493306-19afde93-5649-458b-a977-78fd2d9cc9db.png" />
</details>

<details>
<summary>`shared` changes</summary>
<p>Mostly extension host changes to enable running in VS Code</p>
<img src="https://user-images.githubusercontent.com/37420160/162493387-e59ae845-6e45-4bcf-bbe1-7c103c23c5f4.png" />
</details>

<details>
<summary>`web` changes</summary>
<p>Mostly analytics and import updates</p>
<img src="https://user-images.githubusercontent.com/37420160/162493467-d1a46dbb-9267-425f-a167-e7680cc49ae9.png" />
</details>


## Test plan

- Web app should be unchanged functionally. Automated tests are passing. Manually tested search to ensure that the core flow still works.
  - Run `sg start web-standalone`
  - Execute a search
  - ![web search test](https://user-images.githubusercontent.com/37420160/162491437-aaba18cd-4d23-4519-8f67-0df032c43303.png)

- VS Code extension has been released to users based on this branch multiple times. It is in a working state, confirmed through manual testing before each release. We have added a smoke test to ensure the extension compiles and loads (homepage), and we are in the process of implementing request interception in order to add integration tests.

## App preview:
- [Link](https://sg-web-tjk-vsce-search-ui.onrender.com)
